### DISCUSSION TITLE
Could not locate protobuf definition for response body of firebaselogging

### DISCUSSION THREAD

**Discussion by sanduluca** [invertase/react-native-firebase#6550] [General]
When I connect react native firebase I can see in Flipper this kind of requests
<img width="1002" alt="image" src="https://user-images.githubusercontent.com/56228534/190925663-6a671a9d-3fbc-420a-9743-3126e1f73143.png">

When I install performance & crashlytics there are much more of this. I have no idea why they happen.
Can't find any info on google about this error. Can some help me understang where its coming from ?  What is wrong configured ?

**Comment by mikehardy** [comment]
I don't understand what the problem is - stated differently: what exact use case are you trying to satisfy here?
If it is just idle curiosity for contents of the response you'll probably want to look upstream in firebase-android-sdk / firebase-ios-sdk or similar
If it is "there should be no network traffic", you'll want to read up on the GDPR / user consent items in the documentation and default collection settings in the firebase.json schema section of the docs (like here https://rnfirebase.io/app/json-config) to turn things off

**Comment by sanduluca** [comment]
It's not just for curiosity. And I don't want to turn things off. I want to understand why this error is happening. And as I said when I install performance there are much more of this.

**Reply by mikehardy** [reply]
I see "200" as the return status. So, the SDK is talking to the google servers, and it's going great, the SDK is receiving a response it understands.

Why would flipper have those protobuf defintions? Have you installed them? I'm not even sure where they are.

Doesn't look like an error to me. Looks like "you are using a debugging tool with protobuf inspection if it has the definitions" but you don't have the definitions. Is that actually a problem?

More interestingly is that Flipper isn't even a viable solution at this point - if you are using react-native-firebase v15 you are not on a supported configuration unless you are configured with `use_frameworks` in your Podfile, and Flipper is not compatible with that, so this is either unsupported as not a good config or unsupported as old (meaning: were we to get the protobuf definitions from upstream repo, they'd be out of date anyway)

At it's heart, Firebase is an RPC system using protobuf on the wire if I understand correctly. So if you are really interested in inspecting the RPC calls between the SDKs and the server APIs in the google computer platform cloud, you need to check out the underlying SDKs, the REST API docs etc, and fire up wireshark or similar (since Flipper is a dead-end solution owing to the lack of use_frameworks support) and start unpacking things.

But I'm going to boldly say: the SDK is working. Flipper is showing you it is working (200 response!). Flipper is showing you it is a protobuf RPC call (which it is, good work Flipper). And Flipper is saying "I don't have the protobuf definitions", in the same way that a debugger with no symbols can't decode a stack. That's not an error.

**Reply by mikehardy** [reply]
For a handle on it, if you must for whatever reason unpack these things: https://search.brave.com/search?q=google+firebase+rpc+protobuf

**Comment by sanduluca** [comment]
Ok I agree that 200 ok shows that the request went well. 
I got your point about the proto definitions.

I am using react-native-firebase `14.12.0`. I didn't configure `use_frameworks`

Now when I connected `@react-native-firebase/crashlytics"` and `@react-native-firebase/perf` I dont receive a http 200 ok to the requests. This requests a happening every couple of seconds  

![image](https://user-images.githubusercontent.com/56228534/191016954-76510e83-06ac-4f2c-94ee-4e337d058378.png)

**Reply by mikehardy** [reply]
This is not a bug. There is not a problem here. You are using a "Backend as a Service" system and there should be no surprise at all that the frontend (your app) is talking to the backend.

How do you think analytics and performance is measured so you can have reports on the backend? There is no magic. The bits are sent over the wire.

**Reply by sanduluca** [reply]
I understand that the analytics should report data to the backend and I was expected that there will be some traffic. 
But what bothers me is that all the requests dont receive a 200 ok response. Knowing that  **RPC is a request–response protocol** and  that in first image we saw a 200 ok this means that something is wrong.

**Reply by mikehardy** [reply]
If you can demonstrate a functionality issue we can continue discussing.
